### PR TITLE
abcl: initial port

### PR DESCRIPTION
**Pull request by easye**
With these changes, abcl-1.7.1-dev will compile and load CLX.

Fixes <https://github.com/armedbear/abcl/issues/240>.

### CODE REVIEW

File: dependent.lisp [hunk 1]
```diff
@@ -622,6 +626,21 @@
...
+  (declare (ignore display whostate timeout))
+  `(threads:with-thread-lock (,lock)
+     ,@body))
+
+#+(or) ;; How to implement a timeout?
+(defmacro holding-lock ((lock display &optional (whostate "CLX wait")
+                              &key timeout)
+                        &body body)
+  (declare (ignore display whostate))
+  (threads:with-thread-lock (lock)))
```
Comment (dkochmanski) [inline dependent.lisp:642]: please remove commented out code block. you may add the question comment in the actual macro above.

as of implementing timeouts, there are two approaches: either your mutex implementation must have such interface, or you interrupt the thread (if you support interrupts) from a thread which has a function more or less like this:

```lisp
(defun interrupter (whom timeout)
   (sleep timeout)
    (when (thread-alive-p whom)
        (interrupt-thread whom (lambda () (release-lock) (signal 'timeout))))
```
the lambda function which releases the lock, to know which lock needs to be released,  must be expanded from holding lock to capture the variable.

File: dependent.lisp [hunk 2]
```diff
@@ -810,7 +838,7 @@
   (defvar *conditional-store-lock*
     (make-process-lock "conditional store"))
   (defmacro conditional-store (place old-value new-value)
-    `(holding-lock (*conditional-store-lock*)
+    `(holding-lock (*conditional-store-lock* nil)
```
Comment (dkochmanski) [inline dependent.lisp:841]: what is the rationale to put verbatim nil here?

Comment (easye) [reply dependent.lisp:834]: Otherwise it doesn't compile as `holding-lock` has two required arguments.

File: dependent.lisp (hunk 1 above)
Comment (easye) [reply dependent.lisp]: > please remove commented out code block. you may add the question comment in the actual macro above.

Removed in <https://github.com/sharplispers/clx/pull/171/commits/cd8fcc929b49e33d35011931ef8e358274acf67c>.

Comment (easye) [reply dependent.lisp]: > as of implementing timeouts, there are two approaches: either your mutex implementation must have such interface, or you interrupt the thread (if you support interrupts) from a thread which has a function more or less like this:
> 
> ```lisp
> (defun interrupter (whom timeout)
>    (sleep timeout)
>     (when (thread-alive-p whom)
>         (interrupt-thread whom (lambda () (release-lock) (signal 'timeout))))
> ```
> 
> the lambda function which releases the lock, to know which lock needs to be released, must be expanded from holding lock to capture the variable.

ACK.

**Comment by easye** [comment]
N.b. this patch just makes CLX compile.  I don't expect it to really work in the current form as it lacks an implementation of `open-x-stream`.  I'm trying to be busy with other work at this time, so not planning to move this further today.

File: dependent.lisp (hunk 2 above)
Comment (dkochmanski) [reply dependent.lisp:834]: right you are

**Comment by dkochmanski** [comment]
thanks. it doesn't step on toes of other implementations, so I'm fine with merging it. thanks!

**Pull request merged**